### PR TITLE
feat(build): enforce formatting of some files.

### PR DESCRIPTION
**Pull request by alexeagle**
Our style guide includes formatting conventions. Instead of wasting time in reviewing PRs discussing things like indenting, and to avoid later deltas to fix bad formatting in earlier commits, we want to enforce these in the build.
The intent in this change is to fail the build as quickly as possible in travis, so those sending a PR immediately know they should run clang-format and update their commit. When running locally, we want users to know about formatting, but they may not want to act on it immediately, until they are done working. For this reason, it is only a warning outside of the continuous build.
This is done by having a check-format task which should run on most local builds, and an enforce-format task only run by travis.

### CODE REVIEW

File: tools/broccoli/traceur/index.ts [hunk 1]
```diff
@@ -37,9 +37,9 @@ class TraceurFilter extends Writer {
                 // TODO: we should fix the sourceMappingURL written by Traceur instead of overriding
                 // (but we might switch to typescript first)
                 var mapFilepath = filepath.replace(/\.\w+$/, '') + this.destSourceMapExtension;
-                result.js = result.js + `\n//# sourceMappingURL=./${path.basename(mapFilepath)}`;
+                result.js = result.js + `\n  //# sourceMappingURL=./${path.basename(mapFilepath)}`;
 
-                var destFilepath = filepath.replace(/\.\w+$/, this.destExtension);
+                            var destFilepath = filepath.replace(/\.\w+$/, this.destExtension);
```
Comment (alexeagle) [inline tools/broccoli/traceur/index.ts]: looks like a clang-format bug?

Comment (mprobst) [inline tools/broccoli/traceur/index.ts]: Yes, will TAL.

File: gulpfile.js [hunk 2]
```diff
@@ -508,9 +508,22 @@ gulp.task('build/format.dart', rundartpackage(gulp, gulpPlugins, {
   args: CONFIG.formatDart.args
 }));
 
+function doCheckFormat() {
+  return gulp.src(['Brocfile*.js', 'modules/**/*.ts', 'tools/**/*.ts', '!**/typings/**/*.d.ts'])
```
Comment (mprobst) [inline gulpfile.js]: Why no the typings?

File: gulpfile.js [hunk 3]
```diff
@@ -508,9 +508,22 @@ gulp.task('build/format.dart', rundartpackage(gulp, gulpPlugins, {
...
-  return gulp.src(['Brocfile*.js', 'modules/**/*.ts', '!**/typings/**/*.d.ts'])
-      .pipe(format.checkFormat('file'));
+  return doCheckFormat().on('warning', function(e) {
+    console.log("NOTE: this will be promoted to an ERROR in the continuous build");
+  });
+});
+
+gulp.task('enforce-format', function() {
+  return doCheckFormat().on('warning', function(e) {
+    console.log("Failing the build because files are not formatted according to style rules in .clang-format");
```
Comment (mprobst) [inline gulpfile.js]: This seems a bit wordy, maybe just "Files are not formatted with clang-format"?

File: tools/broccoli/traceur/index.ts (hunk 1 above)
Comment (alexeagle) [inline tools/broccoli/traceur/index.ts]: For now, changed this back to single quotes to work around

File: gulpfile.js (hunk 2 above)
Comment (alexeagle) [inline gulpfile.js]: Because the typings/ directories are populated from tsd

File: gulpfile.js (hunk 3 above)
Comment (alexeagle) [inline gulpfile.js]: done

**Comment by alexeagle** [comment]
Hmm, clang-format doesn't work on the travis arch:

https://travis-ci.org/angular/angular/jobs/58104564#L809

/home/travis/build/angular/angular/node_modules/gulp-clang-format/node_modules/clang-format/bin/linux_x64/clang-format: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.17' not found (required by /home/travis/build/angular/angular/node_modules/gulp-clang-format/node_modules/clang-format/bin/linux_x64/clang-format)

**Comment by mprobst** [comment]
LGTM, assuming travis likes it.

**Comment by mprobst** [comment]
Maybe we can build a statically linked clang-format binary?

**Comment by mprobst** [comment]
Closing & Reopening to trigger a re-build, hopefully with a working clang-format.

**Comment by alexeagle** [comment]
Nice fix, ready to merge?

**Comment by mprobst** [comment]
LGTM.

**Comment by alexeagle** [comment]
merged in daf0f472b3a37d3dc158c0e6bb5c76035bc67037
